ee/clickhouse/queries/experiments/funnel_experiment_result.py

```diff
@@ -0,0 +1,100 @@
+import dataclasses
+from datetime import datetime
+from typing import List, Optional, Tuple, Type
+
+from numpy.random import default_rng
+from rest_framework.exceptions import ValidationError
+
+from ee.clickhouse.queries.funnels import ClickhouseFunnel, funnel
+from posthog.models.filters.filter import Filter
+from posthog.models.team import Team
+
+
+@dataclasses.dataclass
+class Variant:
+    name: str
+    success_count: int
+    failure_count: int
+
+
+SIMULATION_COUNT = 100_000
+
+
+class ClickhouseFunnelExperimentResult:
+    def __init__(
+        self,
+        filter: Filter,
+        team: Team,
+        feature_flag: str,
+        experiment_start_date: datetime,
+        experiment_end_date: Optional[datetime] = None,
+        funnel_class: Type[ClickhouseFunnel] = ClickhouseFunnel,
+    ):
+
+        breakdown_key = f"$feature/{feature_flag}"
+
+        query_filter = filter.with_data(
+            {
+                "date_from": experiment_start_date,
+                "date_to": experiment_end_date,
+                "breakdown": breakdown_key,
+                "breakdown_type": "event",
+                "properties": [
+                    {"key": breakdown_key, "value": ["control", "test"], "operator": "exact", "type": "event"}
+                ],
+                # don't rely on properties used to set filters, taken over by $feature/X event properties
+                # instead, filter by prop names we expect to see: the variant values
+            }
+        )
+        self.funnel = funnel_class(query_filter, team)
+
+    def get_results(self):
+        funnel_results = self.funnel.run()
+        variants = self.get_variants(funnel_results)
+
+        probability = self.calculate_results(variants)
+
+        return {"funnel": funnel_results, "probability": probability}
+
+    def get_variants(self, funnel_results):
+        variants = []
+        for result in funnel_results:
+            total = sum([step["count"] for step in result])
+            success = result[-1]["count"]
+            failure = total - success
+            breakdown_value = result[0]["breakdown_value"][0]
```

Comment:
Will this work for multi-breakdowns? Seems really really suspect.

Reply:
Nope, won't work for multi-breakdowns, but since we're creating the funnel ourselves, we know that there isn't a multi-breakdown.

Hmm, this is valuable feedback, I've been wrangling with this for a few days now, so much that I've forgotten the fresh-eyes look. Will add more documentation at the top of the class to explain what's going on.

I didn't want to move this yet to use `breakdown` vs `breakdowns` as there seem to be some inconsistencies / bugs around this. Will update once we get rid of that tech debt.